### PR TITLE
Add aim and aimrecords recipes

### PR DESCRIPTION
**Pull request by psortos**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by psortos** [comment]
@conda-forge/staged-recipes ready for review, thank you!

File: recipes/aimrecords/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - pip
+    - python >=3.6
+    - base58
+    - pytest
+    - ipdb
```
Comment (chrisburr) [inline recipes/aimrecords/meta.yaml:23]: These are likely only run dependencies

File: recipes/aim/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+requirements:
+  host:
+    - pip
+    - python
+    - googlesearch
+    - protobuf
+    - anytree
+    - pyrser
+    - psutil
```
Comment (chrisburr) [inline recipes/aim/meta.yaml:24]: These are likely only run dependencies

File: recipes/aim/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+package:
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: https://github.com/aimhubio/{{ name }}/archive/v{{ version }}.tar.gz
+  sha256: af4f06e7ecc3b30537370e732f172670588de580eaa073c7eeed248dec63f248
+
+build:
+  number: 0
```
Comment (chrisburr) [inline recipes/aim/meta.yaml:13]: This looks like it can be `noarch: python`:

```suggestion
  noarch: python
  number: 0
```

Comment (chrisburr) [reply recipes/aim/meta.yaml:13]: That happens when the recipe is missing console_scripts entrypoints, see:

* https://conda.io/projects/conda-build/en/latest/resources/define-metadata.html#python-entry-points
* https://github.com/aimhubio/aim/blob/7d31a0feeb2d958cbaf5ed34e410737168a28c3d/setup.py#L133

File: recipes/aim/meta.yaml (hunk 2 above)
Comment (chrisburr) [reply recipes/aim/meta.yaml:24]: Sorry, looks like they are actually needed as the package imports itself as build time: https://github.com/aimhubio/aim/blob/7d31a0feeb2d958cbaf5ed34e410737168a28c3d/setup.py#L7

File: recipes/aimrecords/meta.yaml (hunk 1 above)
Comment (psortos) [reply recipes/aimrecords/meta.yaml]: tested this out and base58 is required, but the others aren't

File: recipes/aim/meta.yaml (hunk 2 above)
Comment (psortos) [reply recipes/aim/meta.yaml]: just retested, I received build failures for each google/protobuf, anytree, pyrser and psutil when I remove them from host

File: recipes/aim/meta.yaml (hunk 3 above)
Comment (psortos) [reply recipes/aim/meta.yaml:14]: fixed! Thanks for the heads up

**Comment by psortos** [comment]
@chrisburr ready for re-review, thank you!

**Comment by psortos** [comment]
@mariusvniekerk 

Just updated this to use tarballs from pypi

**Comment by psortos** [comment]
@conda-forge/staged-recipes ready for review!

File: recipes/aimrecords/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+requirements:
+  host:
+    - pip
+    - python >=3.6
+    - base58
+  run:
+    - python >=3.6
+    - pytest
+    - ipdb
+    - base58
```
Comment (xhochy) [inline recipes/aimrecords/meta.yaml:26]: Looking at upstream `pytest` and `ipdb` aren't necessary but `base58` needs to be pinned very strictly: https://github.com/aimhubio/aimrecords/blob/9a6d1cfe3877de633d227afb221e185eb9fc7c4a/setup.py#L30
```suggestion
    - base58 2.0.1
```

File: recipes/aim/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+requirements:
+  host:
+    - pip
+    - python >=3.6
+    - googlesearch
+    - protobuf
+    - anytree
+    - pyrser
+    - psutil
+  run:
```
Comment (xhochy) [inline recipes/aim/meta.yaml:28]: The requirements upstream are much longer and seem to be pinned very strictly: https://github.com/aimhubio/aim/blob/7d31a0feeb2d958cbaf5ed34e410737168a28c3d/setup.py#L39

Comment (psortos) [reply recipes/aim/meta.yaml:23]: gotcha, just added more requirements by using grayskull this time

Comment (psortos) [reply recipes/aim/meta.yaml:23]: @xhochy might have to WIP this, looks like we might not have `pysqlite3` in the conda-forge and the specific version of  `flask-migrate 2.7.0` either

File: recipes/aimrecords/meta.yaml (hunk 4 above)
Comment (psortos) [reply recipes/aimrecords/meta.yaml]: @xhochy I was able to the pins + additional requirements

File: recipes/aim/meta.yaml (hunk 5 above)
Comment (psortos) [reply recipes/aim/meta.yaml:28]: @xhochy I was able to the pins + additional requirements

**Pull request merged**